print('hola')

$$X= W \Sigma V^T$$

In [46]:
import numpy as np 
X=np.random.rand(7,5)
U,S,V=np.linalg.svd(X,full_matrices=True) #Full SVD
V

array([[-0.37296215, -0.49253591, -0.55189698, -0.47659979, -0.29422777],
       [ 0.64439137,  0.21064135, -0.56564977, -0.2932758 ,  0.36663283],
       [-0.08598237,  0.74849212, -0.13866305, -0.16160479, -0.62211175],
       [-0.53832208,  0.27495873, -0.50278746,  0.47638438,  0.39353492],
       [-0.38532502,  0.27784139,  0.3216013 , -0.65862263,  0.48694725]])

In [60]:
#Hora de importar datos de ENSA :00
import pandas as pd 
import os 


# Ruta del directorio donde están tus archivos .PRN [Energías]
input_dir = r"C:\Users\nasca\Desktop\POYECTO CM 2025\lecturas_ener_procesadas"

# Nombre (y ruta) para el archivo de salida (CSV final)
#output_file = r"C:\Users\nasca\Desktop\ONE DRIVE CM\DATA POWER BI CLIENTES LIBRES\ENERGÍAS 2025\ENERO 25\combine_ENE25.csv"

# Obtenemos la lista de archivos que terminen con .prn
prn_files = [f for f in os.listdir(input_dir)] #Lista de Compresión(comprhensive list)
#f.lower() : convierte a minúscula el nombre
#ends.with('.prn'):identifica los archivos prn

# Lista donde almacenaremos temporalmente los DataFrames
dfs = []  #Una vez que se almecena debería eliminarse creo

for prn_filename in prn_files:
    file_path = os.path.join(input_dir, prn_filename) #Necesario para una ruta segura a los archivos PRN, tanto en Windows como Mac
    
    # Leemos cada PRN con pandas
    # Ajustar:
    #   encoding si requieres un encoding particular (ej. 'latin1', 'utf-8', etc.)
    df = pd.read_csv(file_path, 
                     sep='\t',     #Separado por tabulaciones(\t), hay más... 
                     header=0,     # Cambiar a None si el archivo NO tiene header
                     engine="c"    # Usa 'python' si los PRNs tienen spacing variable
                    )
    
    # Aquí combinamos un archivo PRN a otro 
    #OJO : Aquí solo está una lista de listas lista_de_listas= [ [prn_filename1],[prn_filename2],...]
    dfs.append(df)

c_df = pd.concat(dfs, ignore_index=True, sort=False)
# Concatenamos todos los DataFrames con "sort=False" no ordena alfebeticamente las columnas.
# ignore_index=True : ignora los índices de cada df, para evitar indices duplicados.
# Las columnas que no estén presentes en algún archivo se rellenan con NaN en la concatenación.

# Guardamos el resultado en un CSV (sin índice)
#c_df.to_csv(output_file, index=False, encoding="utf-8")

print("¡Archivos combinados con éxito! Archivo resultante, muy bien ")

# Duda aqui parece que concatena listas no DFs ....


¡Archivos combinados con éxito! Archivo resultante, muy bien 


In [83]:
# Primero energías
#Datos del mes de septiembre 2025 de SOLO ENERGÍAS
c_df_ener=c_df[['Account:','kWh-Del']]

print(f"Hay {c_df_ener['Account:'].nunique()} clientes")


Hay 1277 clientes


In [85]:
# Limpieza de datos hay clientes que tienen lecturas incompletas 
c_df_ener['Account:'].value_counts()

Account:
41144411    2976
25000010    2976
25000074    2976
25000181    2976
25002088    2976
            ... 
26407527    1083
26458894     988
35893437     925
26248912     922
35707003       1
Name: count, Length: 1277, dtype: int64

In [92]:
c_df_ener.groupby('Account:')['kWh-Del'].transform('size').eq(2976)

0          True
1          True
2          True
3          True
4          True
           ... 
3772539    True
3772540    True
3772541    True
3772542    True
3772543    True
Name: kWh-Del, Length: 3772544, dtype: bool

## Con Energías
$$\bar{X} =
\begin{bmatrix}
\mid & \mid &        & \mid \\
\mathbf{x}_1 & \mathbf{x}_2 & \cdots & \mathbf{x}_{k=1277} \\
\mid & \mid &        & \mid
\end{bmatrix} $$
- Donde cada $x_k$ va a representa un cliente con lecuturas de demanda en kW (kW-Del)
- Cada fila $x^{(i)} \in \mathbb{R}^{1277}$  representa una lectura 15-minutal

In [ ]:
# Doy forma a mi dataframe y lo convierto a dato de numpy para procesar el algoritmo de SVD


Account:
41144411    2976
25000010    2976
25000074    2976
25000181    2976
25002088    2976
            ... 
26407527    1083
26458894     988
35893437     925
26248912     922
35707003       1
Name: count, Length: 1277, dtype: int64